zeebe/backup/src/test/java/io/camunda/zeebe/backup/management/ConcurrentBackupCompactionTest.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.backup.management;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.primitive.partition.PartitionId;
+import io.atomix.primitive.partition.PartitionMetadata;
+import io.atomix.raft.impl.LogCompactor;
+import io.atomix.raft.metrics.RaftServiceMetrics;
+import io.atomix.raft.partition.RaftPartition;
+import io.atomix.raft.storage.log.RaftLog;
+import io.atomix.utils.concurrent.ThreadContext;
+import io.camunda.zeebe.journal.JournalMetaStore;
+import io.camunda.zeebe.journal.Segments;
+import io.camunda.zeebe.journal.file.SegmentedJournal;
+import io.camunda.zeebe.scheduler.ActorScheduler;
+import io.camunda.zeebe.scheduler.SchedulingHints;
+import io.camunda.zeebe.snapshots.PersistedSnapshot;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotStore;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotStoreImpl;
+import io.camunda.zeebe.test.util.AutoCloseableRule;
+import io.camunda.zeebe.util.FileUtil;
+import io.camunda.zeebe.util.buffer.DirectBufferWriter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.awaitility.Awaitility;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ExtendWith(MockitoExtension.class)
+public class ConcurrentBackupCompactionTest {
```

Comment:
:x: We try and write all new tests using Junit 5. Some tests are still with Junit 4 _if_ they use things (e.g. rules) which are not yet usable for Junit 5. In this case, we have an `AutoCloseResources` extension, and there is a `@TmpDir` annotation for Junit 5. There's also a Mockito extension. So I don't see any reason not to write the test as Junit 5, unless I missed something.

Reply:
Sorry was looking for `AutoCloseResources` but I only found `AutoCloseableRule` so I thought it would be the right one. Will move to what you suggested.